src/AWS.Deploy.Orchestration/DeploymentManifestEngine.cs

```diff
@@ -0,0 +1,222 @@
+                var lastDeployedStack = deploymentManifestModel?.LastDeployedStacks
+                    .FirstOrDefault(x => x.AWSAccountId.Equals(_session.AWSAccountId) && x.AWSRegion.Equals(_session.AWSRegion));
+
+                if (deploymentManifestModel != null && lastDeployedStack != null)
+                {
+                    lastDeployedStack.Stacks.Remove(stackName);
+
+                    await WriteDeploymentManifestFile(deploymentManifestModel);
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new FailedToUpdateDeploymentManifestFileException($"Failed to update the deployment manifest file " +
+                    $"to delete the stack '{stackName}'.", ex);
+            }
+        }
+
+        /// <summary>
+        /// This method deserializes the deployment-manifest file and returns a list of absolute paths of directories at which different CDK
+        /// deployment projects are stored. The custom recipe snapshots are stored in this directory.
+        /// </summary>
+        /// <returns> A list containing absolute directory paths for CDK deployment projects.</returns>
+        public async Task<List<string>> GetRecipeDefinitionPaths()
+        {
+            var recipeDefinitionPaths = new List<string>();
+            var targetApplicationDirectoryPath = _directoryManager.GetDirectoryInfo(_targetApplicationFullPath).Parent.FullName;
+
+            var deploymentManifestModel = await GetDeploymentManifest();
+            if (deploymentManifestModel == null)
+                return recipeDefinitionPaths;
+
+            foreach (var entry in deploymentManifestModel.DeploymentManifestEntries)
+            {
+                var saveCdkDirectoryRelativePath = entry.SaveCdkDirectoryRelativePath;
+                if (string.IsNullOrEmpty(saveCdkDirectoryRelativePath))
+                    continue;
+
+                var saveCdkDirectoryAbsolutePath = _directoryManager.GetAbsolutePath(targetApplicationDirectoryPath, saveCdkDirectoryRelativePath);
+
+                if (_directoryManager.Exists(saveCdkDirectoryAbsolutePath))
+                    recipeDefinitionPaths.Add(saveCdkDirectoryAbsolutePath);
+            }
+
+            return recipeDefinitionPaths;
+        }
+
+        /// <summary>
+        /// This method parses the deployment-manifest file into a <see cref="DeploymentManifestModel"/>
+        /// </summary>
+        public async Task<DeploymentManifestModel?> GetDeploymentManifest()
+        {
+            var deploymentManifestFilePath = GetDeploymentManifestFilePath(_targetApplicationFullPath);
+
+            if (!_fileManager.Exists(deploymentManifestFilePath))
+                return null;
+            var manifestFilejsonString = await _fileManager.ReadAllTextAsync(deploymentManifestFilePath);
+            return JsonConvert.DeserializeObject<DeploymentManifestModel>(manifestFilejsonString);
```

Comment:
I think if we fail to deserialize `aws-deployments.json` we should log it and then return an empty `DeploymentManifestModel`. We shouldn't fail deployment because somebody dumb fingered this file of mostly optional information. Especially since if we fail to deserialize the error will manifest in probably confusing ways in the toolkit.

Reply:
Fixed